### PR TITLE
chore: firebase app distribution

### PR DESCRIPTION
**Pull request by brainbicycle**
The type of this PR is: **chore**

This PR resolves [CX-1809]

### Description

Uploads android betas to firebase app distribution as well as google play alpha track. Firebase app distribution will be the typical place for devs to get betas for testing and google play will contain release candidates. The only difference between the two builds should be defaulting to staging in dev betas and defaulting to production in release candidates. This way we can test in betas and confidently promote the corresponding release candidate to the play store.

Implementation:
- Adds firebase app distribution fastlane plugin
- Builds the app twice with the beta flag set to true and false
- Uploads to google play and firebase app distribution
- Reverts earlier promotion change to just promote from google play rather than rebuild

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Use firebase app distribution for android betas - Brian

</details>

[CX-1809]: https://artsyproduct.atlassian.net/browse/CX-1809

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Use firebase app distribution for android betas - Brian)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b0627bd28284b3ba07567235a472e090399767e5
</p>

File: fastlane/Fastfile [hunk 1]
```diff
@@ -170,8 +170,20 @@ lane :ship_beta_android do
     project_dir: "android/",
   )
 
+  firebase_app_distribution(
+    app: ENV["FIREBASE_APP_ID"],
```
Comment (lordkiz) [inline fastlane/Fastfile:174]: @brainbicycle How are we authenticating with firebase? Considering that you haven't passed a cli token. I might be missing something here

Comment (brainbicycle) [reply fastlane/Fastfile:174]: we have a new env var GOOGLE_APPLICATION_CREDENTIALS that points to our google credentials file, this file will be downloaded by our script here: scripts/setup-distribute-linux, doesn't need to be passed explicitly as long as it is in the env file

Comment (lordkiz) [reply fastlane/Fastfile:174]: Awesome!

**Review by lordkiz — APPROVED** [review]
👏

**Pull request merged**